### PR TITLE
Install yamllint in dev docker image

### PR DESCRIPTION
**Pull request by wangkuiyi**
In https://github.com/sql-machine-learning/sqlflow/pull/2256, we removed wangkuiyi/yamlfmt from the pre-commit hook.  We want to replace it with yamllint.  This PR add yamllint.  After its merge, we can add it to the pre-commit hook.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -49,6 +49,7 @@ require (
 	gopkg.in/alecthomas/kingpin.v2 v2.2.6 // indirect
 	gopkg.in/natefinch/lumberjack.v2 v2.0.0
 	gopkg.in/yaml.v2 v2.2.7 // indirect
+	honnef.co/go/tools v0.0.0-20190523083050-ea95bdfd59fc // indirect
```
Comment (tonyyang-svail) [inline go.mod:52]: Why update `go.mod`?

Comment (wangkuiyi) [reply go.mod:52]: I have no idea. In all my recent PRs, `git commit -a` or something else automatically adds this change.

Comment (wangkuiyi) [reply go.mod:52]: @tonyyang-svail I verified on develop branch.  `go generate ./...` introduces this change.

Comment (tonyyang-svail) [reply go.mod:52]: I see. Looks like `linter` depends on it. https://pkg.go.dev/mod/honnef.co/go/tools@v0.0.1-2020.1.4

**Review by tonyyang-svail — APPROVED** [review]
LGTM

**Pull request merged**